### ISSUE TITLE
Accept legacy datafolders for a limited time for backward compatibility

### ISSUE THREAD

**Issue by jcrivenaes** [equinor/fmu-dataio#187]
Historically, output from FMU work flows comes in defined folders. Much of the current ecosystem around is hard-coded on those names, and the proposal here is a `force_folder` key that allow deviance, at least for a temporary time.

For examples, output of volumes currently comes `../share/results/tables`, but many expect `../share/result/volumes`. Hence this can be fulfilled with this key:

`edata.export(voltable_as_pandas, force_folder="..share/results/volumes", ...)` which will override the default setting from dataio.

Usage of `force_folder` shall always be accompanied with a `UserWarning`

Comments @perolavsvendsen ?

**Comment by perolavsvendsen** [comment]
I can see the need (and technically already possible (and done) using the `subfolder` arg).
What is the reference point for this relative folder, and will it work when using fmu-dataio in contexts which is not the ERT RUNPATH?

**Comment by jcrivenaes** [comment]
Yes, RUNPATH if relative, so perhaps the example shall be force_folder="share/results/volumes"

**Comment by perolavsvendsen** [comment]
This will be challenging when context cannot be RUNPATH, i.e when RUNPATH does not exist. Example is when doing pre- or post-processing. In those cases, reference will typically be "casepath".

May be OK if we can live with different behavior of `export` in different contexts. Ref ongoing discussion around #59 and #174.

The existing `subfolder` arg is relative to standard location given current context, I believe. Not intentional, and possibly not tested for all contexts...

**Comment by jcrivenaes** [comment]
Yes subfolder is always relative. As for `RUNPATH` we use the state variable `_runpath` in the code, but it may more generic to use something like `_basepath`, where `_basepath` reflects RUNPATH in current forward models, but may be something else in other contexts?

**Comment by perolavsvendsen** [comment]
Yes... Or, not sure. But handling the context is important. Possibly a prerequisite for a `force_folder` implementation? E.g we need to know, if a relative folder is given, what it is relative to. And this may be different in different contexts. But it may also be fine to do so. Need to test with some examples, perhaps? 

Known contexts:
- In RMS, template-side (special case, not producing valid metadata) 
- In RMS, realization (separate Python dist.)
- In any (?) other FORWARD model, realization
- In hooked workflow, post-process
- In non-hooked wf, post
- In hooked workflow, pre-process
- In non-hooked wf, pre-process

Is it a possible solution to specify filename in the export call? E.g if `save_as` or something similar (full filename) is passed, do basic sanity checks then use the filename. If not (default), construct it? This will outsource the full construction of the filename to the caller if they want to deviate from standard. (Perhaps this is not so far from the actual suggestion, really...)

**Comment by slangeveld** [comment]
forcefolder already implemented.
